pkg/storage/calico/resource.go

```diff
@@ -419,15 +425,31 @@ func (rs *resourceStore) GuaranteedUpdate(
 		}
 		revInt, _ := strconv.Atoi(accessor.GetResourceVersion())
 		updatedRes := updatedObj.(resourceObject)
-		if updatedRes.GetObjectMeta().GetResourceVersion() == "" || revInt < int(curState.rev) {
-			updatedRes.(resourceObject).GetObjectMeta().SetResourceVersion(strconv.FormatInt(curState.rev, 10))
+		if shouldCreateOnUpdate() {
+			// As a precaution, cleanup revision in case it is set by userUpdate function.
+			updatedRes.(resourceObject).GetObjectMeta().SetResourceVersion("")
```

Comment:
Does userUpdate ever actually do this? 

What sort of scenarios might it? If it does, do we want to override what it has done or should we respect it?

Reply:
In reality, I suppose `userUpdate` function should never touch `resource version` as it is populated by backend storage (etcd). However, our CI test (see above comment) actually uses a stored object (which has a valid resource version) to replace the original object. 

Yeah, we should not add code here just to satisfy a wrong `userUpdate` function passed by our CI. We should update CI code to do the right thing.